pwiz_tools/Skyline/TestFunctional/AuditLogValidationTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * Original author: Rita Chupalov <ritach .at. uw.edu>,
+ *                  MacCoss Lab, Department of Genome Sciences, UW
+ *
+ * Copyright 2019 University of Washington - Seattle, WA
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using pwiz.Skyline.Alerts;
+using pwiz.Skyline.Controls.AuditLog;
+using pwiz.Skyline.Model.AuditLog;
+using pwiz.SkylineTestUtil;
+
+namespace pwiz.SkylineTestFunctional
+{
+    [TestClass]
+    public class AuditLogValidationTest : AbstractFunctionalTestEx
+    {
+        [TestMethod]
+        public void TestAuditLogValidation()
+        {
+            TestFilesZip = "TestFunctional/AuditLogValidationTest.zip";
+            RunFunctionalTest();
+
+        }
+
+        protected override void DoTest()
+        {
+            //testing scenarios:
+            // - valid audit log file - opens cleanly
+            var documentFile = TestFilesDir.GetTestPath(@"AuditLogValidationTest/MethodEditClean.sky");
+            WaitForCondition(() => File.Exists(documentFile));
```

Comment:
I don't understand why you're doing "WaitForCondition" instead of just "Assert"

Reply:
Well, this is the first Skyline functional test I wrote. This line was copied from AbstractFunctionalTestEx.OpenDocument() method. I would assume that ZIP extraction might be still in progress when the test starts, hence the wait. If it's not true then maybe the base class should be modified as well.